hcmanager/utils_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hcmanager
+
+import (
+	hwcc "github.com/metal3-io/hardware-classification-controller/api/v1alpha1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	bmoapis "github.com/metal3-io/baremetal-operator/pkg/apis"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/klogr"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("HCManager", func() {
+
+	hostTest := getHosts()
+
+	c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
+	hcManager := NewHardwareClassificationManager(c, klogr.New())
+
+	It("Should fetch BaremetalHosts in ready state and under metal3 namespace", func() {
```

Comment:
you could avoid the repetition here by using table driven tests, for each function you want to test

Reply:
done. @maelk @digambar15 kindly review